### PR TITLE
Fixes learnweb/moodle-mod_moodleoverflow#75

### PR DESCRIPTION
**Pull request by SpaceCafe**
Fixed database issue in adhoc-task `tool_dataprivacy\task\process_data_request_task` by splitting query into two parts due to a never ending database request.

### CODE REVIEW

**Comment by ziegenberg** [comment]
The fix is faster, but for our dataset, it's still taking more than 8 minutes to return an empty set for the first of the two queries. Maybe split it into three or four single queries? Or the alternative query mentioned in the issue comment: https://github.com/learnweb/moodle-mod_moodleoverflow/issues/75#issuecomment-1520099973.

**Comment by geichelberger** [comment]
The alternative query is much harder to understand. I would go for readability and split up the joins as it's done in mod_assign.

File: classes/privacy/provider.php [hunk 1]
```diff
@@ -142,39 +142,36 @@ public static function get_contexts_for_userid(int $userid) : contextlist {
...
+                    SELECT 1 FROM {moodleoverflow_subscriptions} s WHERE s.moodleoverflow = mof.id AND s.userid = :suserid
+                ) OR EXISTS (
+                    SELECT 1 FROM {moodleoverflow_discuss_subs} ds WHERE ds.moodleoverflow = mof.id AND ds.userid = :dsuserid
+                ) OR EXISTS (
+                    SELECT 1 FROM {moodleoverflow_ratings} ra WHERE ra.moodleoverflowid = mof.id AND ra.userid = :rauserid
+                ) OR EXISTS (
+                    SELECT 1 FROM {moodleoverflow_tracking} track WHERE track.moodleoverflowid = mof.id AND track.userid = :tuserid
+                ) OR EXISTS (
+                    SELECT 1 FROM {moodleoverflow_grades} g WHERE g.moodleoverflowid = mof.id AND g.userid = :guserid
+                );";
```
Comment (ziegenberg) [inline classes/privacy/provider.php:161]:
```suggestion
                )";
```
This semicolon might be too much.

**Comment by NinaHerrmann** [comment]
Sorry, I hope I can trust you on this one, as test are not failing :) :+1:

**Pull request merged**